first iteration

rst1989 opened the pull request.

rst1989 committed:
first iteration

shishov1101 commented on main.cpp:
```diff
@@ -0,0 +1,125 @@
+
+    for (Number i = 0; i < row_count; ++i){
+        getline(input, input_string);
+        read_result.push_back(Domain(input_string));
```
В целом тут ты заранее знаешь размер, можно и сразу делать reserve, а дальше emplace back

shishov1101 commented on main.cpp:
```diff
@@ -0,0 +1,125 @@
+};
+
+template<typename Number>
+std::vector<Domain> ReadDomains(std::istream& input, const Number row_count) {
```
Возможно Number лучше передать по константной ссылке. Ну это конечно на подумать

shishov1101 commented on main.cpp:
```diff
@@ -0,0 +1,125 @@
+
+template<typename Iter>
+DomainChecker::DomainChecker(Iter begin, Iter end){
+    while (std::distance(begin, end)){
```
Не думаю, что лучшее решение использовать такое условие в цикле, если ты будешь использовать контейнер с итератором не произвольного доступа, то логарифмическая сложность будет O(n)

shishov1101 reviewed (commented):
Отличная работа. Код хорошо читаем. Есть пара замечаний, которые можно поправить, однако для данной программы они не очень критичны.

rst1989 committed:
Update main.cpp

Добавлена константная ссылка в ReadDomains. 
Изменен конструктор класса DomainChecker

rst1989 committed:
Update main.cpp

Исправлены указанные недостатки

shishov1101 reviewed (approved):
Замечания исправлены, принято
